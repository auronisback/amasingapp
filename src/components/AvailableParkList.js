import React, {Component} from 'react';
import ParkStroke from "./ParkStroke";

/**
 * Component managing the set of available parking lots.
 * */
export default class AvailableParkList extends Component {

  /**
   * Strokes appearance.
   * */
  static STROKE_COLOR = '#777';
  static STROKE_OPACITY = 0.5;

  /**
   * Renders the component.
   * @return The react component
   * */
  render() {
    return (
        <div>
          {this._buildAvailableParks()}
        </div>
    );
  }

  /**
   * Creates elements for the available parking lots.
   * */
  _buildAvailableParks = () => {
    let availableParks = [];
    if(this.props.parks)
      this.props.parks.forEach((elem) => {
        availableParks.push(
            <ParkStroke
                key={elem.id}
                start={elem.start}
                end={elem.end}
                color={AvailableParkList.STROKE_COLOR}
                opacity={AvailableParkList.STROKE_OPACITY}
            />);
      });
    return availableParks;
  }
}