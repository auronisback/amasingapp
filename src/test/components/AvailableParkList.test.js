import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import AvailableParkList from "../../components/AvailableParkList";

Enzyme.configure({adapter: new Adapter()});

describe('<AvailabelParkList>', () => {
  it('Should be empty when no park has been given', () => {
    const list = shallow(<AvailableParkList/>);
    expect(list).toMatchSnapshot();
  });

  it('Should show park lots when given', () => {
    const parks = [
      {
        id: 1,
        start: {lat: 1, lng: 10},
        end: {lat: 100, lng: 1000}
      },
      {
        id: 2,
        start: {lat: 2, lng: 20},
        end: {lat: 200, lng: 2000}
      }
    ];
    const list = shallow(<AvailableParkList parks={parks} />);
    expect(list).toMatchSnapshot();
  });
});