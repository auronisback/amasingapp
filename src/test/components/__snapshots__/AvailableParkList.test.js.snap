// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AvailabelParkList> Should be empty when no park has been given 1`] = `<div />`;

exports[`<AvailabelParkList> Should show park lots when given 1`] = `
<div>
  <ParkStroke
    color="#777"
    end={
      Object {
        "lat": 100,
        "lng": 1000,
      }
    }
    key="1"
    opacity={0.5}
    start={
      Object {
        "lat": 1,
        "lng": 10,
      }
    }
  />
  <ParkStroke
    color="#777"
    end={
      Object {
        "lat": 200,
        "lng": 2000,
      }
    }
    key="2"
    opacity={0.5}
    start={
      Object {
        "lat": 2,
        "lng": 20,
      }
    }
  />
</div>
`;
